### DISCUSSION TITLE
Paddle OCR not working for live video feed

### DISCUSSION THREAD

**Discussion by Adhitya1996** [PaddlePaddle/PaddleOCR#12278] [Q&A]
Issue: I have experimented a lot for several weeks but I am pasting a simple code version to understand if logic is failing at any point. I request the community's help at the earliest. Ofc, Paddle OCR worked well with images and I was able to fetch very good results but I need it to work for live video feed.

Error:
Traceback (most recent call last):
  File "D:\01_IAV-Codes\02_OCR-Models\test_01.py", line 22, in <module>
    frame = draw_ocr(frame, line, font_path=font_path)
  File "C:\Users\I011786\Miniconda\envs\env-paddle\lib\site-packages\paddleocr\tools\infer\utility.py", line 384, in draw_ocr
    scores = [1] * len(boxes)
TypeError: object of type 'NoneType' has no len()

My code:
import cv2
from paddleocr import PaddleOCR, draw_ocr
from PIL import Image, ImageFont

# Start video capture
cap = cv2.VideoCapture(0)
# Use a default font
font_path = ImageFont.load_default()
ocr = PaddleOCR(use_gpu=True, use_angle_cls=True, lang='en') # need to run only once to download and load model into memory
while True:
    # Capture frame-by-frame
    ret, frame = cap.read()
    if not ret:
        break

    # Perform OCR on the frame
    result = ocr.ocr(frame)

    # Display the resulting frame with bounding boxes and detected texts
    if result:
        for line in result:
            frame = draw_ocr(frame, line, font_path=font_path)

    # Only for images
    # for idx in range(len(result)):
    #     res = result[idx]
    #     print("\n Line: \n")
    #     for line in res:
    #         print(line)

    # Display the frame
    cv2.imshow("Video", frame)

    if cv2.waitKey(1) & 0xFF == ord('q'):
        break

cap.release()
cv2.destroyAllWindows()

**Comment by Sunting78** [comment]
Hello, please check if the result is normal. You can print and paste the result here for a look.

**Comment by Adhitya1996** [comment]
Hey, thanks a lot for replying but the result is not printed at all. Issue I guess is with video frames. It worked well for images for me because when images are attached as JPEG or PDF, there are no empty values. But for video frames, it is probably giving out empty arrays and I tried to solve around it but it didn't work for me so far. Hence, this issue: TypeError: object of type 'NoneType' has no len().

Code:

import cv2
from paddleocr import PaddleOCR, draw_ocr
from PIL import Image, ImageFont

# Start video capture
cap = cv2.VideoCapture(0)
# Use a default font
font_path = ImageFont.load_default()
ocr = PaddleOCR(use_gpu=True, use_angle_cls=True, lang='en') # need to run only once to download and load model into memory
while True:
    # Capture frame-by-frame
    ret, frame = cap.read()
    if not ret:
        break

    # Perform OCR on the frame
    result = ocr.ocr(frame)
    print("Result 1")
    print(result)

    # Display the resulting frame with bounding boxes and detected texts
    if result:
        for line in result:
            frame = draw_ocr(frame, line, font_path=font_path)
            print("Result 2")
            print(result)

    # Only for images
    # for idx in range(len(result)):
    #     res = result[idx]
    #     print("\n Line: \n")
    #     for line in res:
    #         print(line)

    # Display the frame
    cv2.imshow("Video", frame)

    if cv2.waitKey(1) & 0xFF == ord('q'):
        break

cap.release()
cv2.destroyAllWindows()

Output:

[2024/04/18 10:25:20] ppocr DEBUG: dt_boxes num : 0, elapsed : 0.9545092582702637
[2024/04/18 10:25:20] ppocr DEBUG: cls num  : 0, elapsed : 0
[2024/04/18 10:25:20] ppocr DEBUG: rec_res num  : 0, elapsed : 0.0
Result 1
[None]
Traceback (most recent call last):
  File "D:\01_IAV-Codes\02_OCR-Models\test_01.py", line 24, in <module>
    frame = draw_ocr(frame, line, font_path=font_path)
  File "C:\Users\I011786\Miniconda\envs\env-paddle\lib\site-packages\paddleocr\tools\infer\utility.py", line 384, in draw_ocr
    scores = [1] * len(boxes)
TypeError: object of type 'NoneType' has no len()

**Comment by UserWangZz** [comment]
It seems that there is no target during the detection stage, which leads to an error in obtaining the length of the boxes. You can check whether the OCR detection model parameter configuration is correct, or whether the current frame contains text targets.
